File: gestionart_backend_ruben_y_jessica/api/src/main/java/com/gestionart/api/domain/models/Pedido.java
package com.gestionart.api.domain.models;

import java.time.LocalDateTime;
import java.util.List;

import com.gestionart.api.domain.enums.EstadoPedido;

public class Pedido {
    private Long id;
    private Long idComprador;
    private LocalDateTime fechaCreacion;
    private LocalDateTime fechaConfirmacion;
    private EstadoPedido estado;
    private List<LineaPedido> lineas;

    public Pedido() {
    }

    public Pedido(Long id,Long idComprador,LocalDateTime fechaCreacion,LocalDateTime fechaConfirmacion,EstadoPedido estado,List<LineaPedido> lineas) {
        this.id = id;
        this.idComprador = idComprador;
        this.fechaCreacion = fechaCreacion;
        this.fechaConfirmacion = fechaConfirmacion;
        this.estado = estado;
        this.lineas = lineas;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getIdComprador() {
        return idComprador;
    }

    public void setIdComprador(Long idComprador) {
        this.idComprador = idComprador;
    }

    public LocalDateTime getFechaCreacion() {
        return fechaCreacion;
    }

    public void setFechaCreacion(LocalDateTime fechaCreacion) {
        this.fechaCreacion = fechaCreacion;
    }

    public LocalDateTime getFechaConfirmacion() {
        return fechaConfirmacion;
    }

    public void setFechaConfirmacion(LocalDateTime fechaConfirmacion) {
        this.fechaConfirmacion = fechaConfirmacion;
    }

    public EstadoPedido getEstado() {
        return estado;
    }

    public void setEstado(EstadoPedido estado) {
        this.estado = estado;
    }

    public List<LineaPedido> getLineas() {
        return lineas;
    }

    public void setLineas(List<LineaPedido> lineas) {
        this.lineas = lineas;
    }
    
}

File: gestionart_frontend_ruben_y_jessica/lib/data/models/Pedido.dart


import 'package:gestionart_frontend_ruben_y_jessica/data/enums/EstadoPedido.dart';
import 'package:gestionart_frontend_ruben_y_jessica/data/models/LineaPedido.dart';

class Pedido { //Clase pedido
  final int id;
  final int idComprador;
  final int idVendeodr;
  final Estadopedido estado;
  final List<Lineapedido> lineas;

  Pedido({required this.id, required this.idComprador, required this.idVendeodr, required this.estado, required this.lineas});

  int getId() {
    return id;
  }
  int getIdComprador() {
    return idComprador;
  }
  int getIdVendedor() {
    return idVendeodr;
  }
  Estadopedido getEstado() {
    return estado;
  }
  List<Lineapedido> getLineas() {
    return lineas;
  }

  factory Pedido.fromJson(Map<String, dynamic> json) {
    return Pedido(
      id: json['id'] ?? 0,
      idComprador: json['idComprador'] ?? 0,
      idVendeodr: json['idVendedor'] ?? 0,
      estado: Estadopedido.values.firstWhere((e) => e.toString().split('.').last == 'Estadopedido.' + (json['estado'] ?? ''), orElse: () => Estadopedido.values.first),
      lineas: ((json['lineas'] as List?) ?? []).map((json) => Lineapedido.fromJson(json)).toList()
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id ?? 0,
      'idComprador': idComprador ?? 0,
      'idVendedor': idVendeodr ?? 0,
      'estado': estado.toString().split('.').last ?? Estadopedido.CANCELADO,
      'lineas': (lineas ?? []).map((linea) => linea.toJson()).toList()
    };
  }

   Pedido copyWith({
    int? id,
    int? idComprador,
    int? idVendeodr,
    Estadopedido? estado,
    List<Lineapedido>? lineas,
  }) {
    return Pedido(
      id: id ?? this.id,
      idComprador: idComprador ?? this.idComprador,
      idVendeodr: idVendeodr ?? this.idVendeodr,
      estado: estado ?? this.estado,
      lineas: lineas ?? this.lineas,
    );
  }


}
